### PR TITLE
 fix (core): Cancel command or not allow elevation when tile's elevation level is superior

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
* Not allow parent's elevation when tile's elevation level is superior
* Cancel command or not allow elevation when tile's elevation level is superior

## Motivation and Context
There's conflict between with two elevation layers
Second elevation layer can be resolved before the first layer
Cancel command if target level is lower or equal to current elevation level

### CODE REVIEW

File: src/Process/LayeredMaterialNodeProcessing.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ function initNodeImageryTexturesFromParent(node, parent, layer) {
 
 function initNodeElevationTextureFromParent(node, parent, layer) {
     // inherit parent's elevation texture
-    if (parent.material && parent.material.getElevationLayerLevel() > EMPTY_TEXTURE_ZOOM) {
+    if (parent.material && parent.material.getElevationLayerLevel() > node.material.getElevationLayerLevel()) {
```
Comment (peppsac) [inline src/Process/LayeredMaterialNodeProcessing.js]: Is this change really needed?

(when would `initNodeElevationTextureFromParent` with node's elevation level not being EMPTY_TEXTURE_ZOOM)

File: src/Process/LayeredMaterialNodeProcessing.js [hunk 2]
```diff
@@ -117,6 +117,14 @@ function refinementCommandCancellationFn(cmd) {
         return false;
     }
 
+    // There's conflict between with two elevation layers
```
Comment (peppsac) [inline src/Process/LayeredMaterialNodeProcessing.js]: You could reword this comment:
```js
// Cancel the command if the tile already has a better texture.
// This is only needed for elevation layers, because we may have several
// concurrent layers but we can only use one texture.
```

File: src/Process/LayeredMaterialNodeProcessing.js [hunk 3]
```diff
@@ -352,6 +360,15 @@ export function updateLayeredMaterialNodeElevation(context, layer, node, force)
                 return;
             }
 
+            // There's conflict between two elevation layers
```
Comment (peppsac) [inline src/Process/LayeredMaterialNodeProcessing.js]:
```js
// Do not apply the new texture if its level is < than the current one.
// This is only needed for elevation layers, because we may have several
// concurrent layers but we can only use one texture.
```

**Review by peppsac — CHANGES_REQUESTED** [review]

File: src/Process/LayeredMaterialNodeProcessing.js (hunk 1 above)
Comment (gchoqueux) [reply src/Process/LayeredMaterialNodeProcessing.js]: it's precisely this case that I avoid.
Because with two elevation layers we passe twice in `initNodeElevationTextureFromParent`

File: src/Process/LayeredMaterialNodeProcessing.js (hunk 2 above)
Comment (gchoqueux) [reply src/Process/LayeredMaterialNodeProcessing.js]: ok, thanks

File: src/Process/LayeredMaterialNodeProcessing.js (hunk 3 above)
Comment (gchoqueux) [reply src/Process/LayeredMaterialNodeProcessing.js]: ok, thanks

File: src/Process/LayeredMaterialNodeProcessing.js (hunk 1 above)
Comment (peppsac) [reply src/Process/LayeredMaterialNodeProcessing.js]: Oh right. Then modify the comment above because it's not obvious.
Seomthing like:
```js
      // Inherit parent's elevation texture. Note that contrary to color layers the elevation level of the 
      // node might not be EMPTY_TEXTURE_ZOOM in this init function. That's because we can have
      // multiple elevation layers (thus multiple calls to initNodeElevationTextureFromParent) but a given
      // node can only use 1 elevation texture
```

**Review by peppsac — APPROVED** [review]
(please see https://github.com/iTowns/itowns/pull/570/files#r150302965 before merging)

Thanks.

**Comment by gchoqueux** [comment]
ok
